app/controllers/products_controller.rb

```diff
@@ -1,28 +1,48 @@
-class ProductsController < ApplicationController 
-    def index
-        @products= Product.all
-        @order_item = current_order.order_items.new
-    end
+# Product controller
+class ProductsController < ApplicationController
+  def index
+    @products = Product.all
+    @order_item = current_order.order_items.new
+  end
+
+  def new
+    @product = Product.new
+  end
 
-    def new
-        @product = Product.new
+  def create
+    @product = Product.new(product_params)
+    if @product.save
+      @product.avatar.attach(params[:product][:avatar])
+      redirect_to root_path
+    else
+      redirect_to new_product_path
     end
+  end
 
-    def create
-        @product = Product.new(product_params)
-        if @product.save
-            @product.avatar.attach(params[:product][:avatar])
-            redirect_to root_path
-        else
-            redirect_to new_product_path
-        end
+  def edit
+    @product = Product.find(params[:id])
```

Comment:
you can make a `set_product` method - since this is duplicated in the update and delete methods